File: converting-JavaScript-to-Python/parse-flag.py
import argparse

parser = argparse.ArgumentParser(
    prog="count-containing-words",
    description="Counts words in a file that contain a particular character",
)

parser.add_argument("-c", "--char", help="The character to search for", default="e")
parser.add_argument("path", help="The file to search")

args = parser.parse_args()
print(args)

with open(args.path, "r") as f:
    content = f.read()

print(content);

splitted_content = content.split(" ");
print(splitted_content);

filtered_content = list(filter(lambda x: args.char in x, splitted_content));

counted_filter = len(filtered_content);
print(filtered_content);
print(counted_filter);

File: converting-JavaScript-to-Python/parse-flag.js
import { program } from "commander";
import { promises as fs } from "node:fs";

program
    .name("count-containing-words")
    .description("Counts words in a file that contain a particular character")
    .option("-c, --char <char>", "The character to search for", "e")
    .argument("<path>", "The file path to process");

program.parse();

const argv = program.args;
console.log(argv);
const path = argv[0];
console.log(path);
const char = program.opts().char;
console.log(char);

const content = await fs.readFile(path, "utf-8");
console.log(content);

console.log(content.split(" ").filter((word) => word.includes(char)).length);